### ISSUE TITLE
Tests fail on Windows

### ISSUE THREAD

**Issue by Kolaru** [JuliaIntervals/IntervalArithmetic.jl#140]
When I use Windows 10 with Julia version 0.6.2 (x86_64-w64-mingw32) and the master branch of the package, the tests fail with the following message:

```julia
julia> Pkg.test("IntervalArithmetic")
INFO: Computing test dependencies for IntervalArithmetic...
INFO: No packages to install, update or remove
INFO: Testing IntervalArithmetic
Constructing intervals: Test Failed
  Expression: @interval(1.0e300) == Interval(9.999999999999999e299, 1.0e300)
   Evaluated: Interval(1.0e300, 1.0e300) == Interval(9.999999999999999e299, 1.0e300)
Stacktrace:
 [1] macro expansion at C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\interval_tests\construction.jl:39 [inlined]
 [2] macro expansion at .\test.jl:860 [inlined]
 [3] anonymous at .\<missing>:?
Constructing intervals: Test Failed
  Expression: @interval(-1.0e307) == Interval(-1.0000000000000001e307, -1.0e307)
   Evaluated: Interval(-1.0e307, -1.0e307) == Interval(-1.0000000000000001e307, -1.0e307) Stacktrace:
 [1] macro expansion at C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\interval_tests\construction.jl:40 [inlined]
 [2] macro expansion at .\test.jl:860 [inlined]
 [3] anonymous at .\<missing>:?
Constructing intervals: Test Failed
  Expression: a == @floatinterval("0.1")
   Evaluated: Interval(0.09999999999999999, 0.1) == Interval(0.1, 0.1)
Stacktrace:
 [1] macro expansion at C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\interval_tests\construction.jl:103 [inlined]
 [2] macro expansion at .\test.jl:860 [inlined]
 [3] anonymous at .\<missing>:?
Test Summary:          | Pass  Fail  Total
Constructing intervals |   82     3     85
ERROR: LoadError: LoadError: LoadError: Some tests did not pass: 82 passed, 3 failed, 0 errored, 0 broken.
while loading C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\interval_tests\construction.jl, in expression starting on line 7
while loading C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\interval_tests/intervals.jl, in expression starting on line 3
while loading C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\runtests.jl, in expression starting on line 11
==============================[ ERROR: IntervalArithmetic ]===============================


failed process: Process(`'C:\Users\Kolaru\AppData\Local\Julia-0.6.2\bin\julia.exe' -Cx86-64 '-JC:\Users\Kolaru\AppData\Local\Julia-0.6.2\lib\julia\sys.dll' --compile=yes --depwarn=yes --check-bounds=yes --code-coverage=none --color=yes --compilecache=yes 'C:\Users\Kolaru\.julia\v0.6\IntervalArithmetic\test\runtests.jl'`, ProcessExited(1)) [1]

==========================================================================================

INFO: No packages to install, update or remove
ERROR: IntervalArithmetic had test errors
```

This may be related to #135.

**Comment by dpsanders** [comment]
Ouch. I have no idea what could be causing this, and have no way of testing it; we also don't have Appveyor set up. (If you know how to do that, it would be very welcome!)

Could you maybe try with a previous version of IntervalArithmetic.jl? Presumably one of the recent changes led to this.

**Comment by dpsanders** [comment]
Maybe even a `git bisect`.

**Comment by dpsanders** [comment]
Confirmed by Appveyor:

https://ci.appveyor.com/project/dpsanders/intervalarithmetic-jl/build/1.0.1/job/i2n9tiy6ts6k75d8

Apparently it should be possible to use Appveyor itself to investigate: https://www.appveyor.com/docs/how-to/rdp-to-build-worker/

**Comment by Kolaru** [comment]
I have bisected this with my machine to commit 2ed8164, but the failing tests were actually added by this commit so it is not yet very conclusive.

I will continue to investigate.

**Comment by dpsanders** [comment]
Thanks for looking at this!

Doing
```
julia> @macroexpand @floatinterval("0.1")
:((IntervalArithmetic.Interval)((IntervalArithmetic.atomic)((IntervalArithmetic.Interval){Float64}, "0.1")))
```
shows that it's calling the (new) `atomic` function, which in turn calls `parse`:
```

```
julia> @format full
6

julia> parse(Interval{Float64}, "0.1")
Interval(0.09999999999999999, 0.1)
```
I presume this `parse` call is not returning the correct result on Windows then?

This in turn calls
```
julia> parse(Float64, "0.1", RoundUp)
0.1

julia> parse(Float64, "0.1", RoundDown)
0.09999999999999999
```
which in turn call
```
julia> T = Float64
       setrounding(T, RoundDown) do
               parse(T, "0.1")
           end
0.09999999999999999
```
which are Base Julia features, so we may have found a bug there?
It calls `tryparse`, which then calls a C function :/

Hmm, alternatively the problem is to do with trying to do this with the rounding mode.
Maybe we should be using `BigFloat`s for this instead.

**Comment by dpsanders** [comment]
```
julia> setprecision(53)
53

julia>
       setrounding(BigFloat, RoundDown) do
               parse(BigFloat, "0.1")
           end
9.9999999999999992e-02

julia> Float64(ans)
0.09999999999999999
```

**Comment by dpsanders** [comment]
```jl
julia> function myparse(s, r)
       setprecision(53) do
           setrounding(BigFloat, r) do
                      parse(BigFloat, s)
                  end
       end |> Float64
       end
myparse (generic function with 1 method)

julia> myparse("0.1", RoundDown)
0.09999999999999999
```

**Comment by dpsanders** [comment]
OK, thanks very much for looking into this. Would you like to raise this as
an issue on the JuliaLang repo, or shall I?

Meanwhile let's just add in the custom parse function. I'm happy to do
that.

**Comment by Kolaru** [comment]
Well since you can not easily reproduce the bug it makes more sens that I open the issue. Here it is:

https://github.com/JuliaLang/julia/issues/26934

**Comment by dpsanders** [comment]
Great, thanks!

**Comment by lbenet** [comment]
The following is a suggestion for windows despite the rounding problem: for windows  use `:accurate` as the default rounding mode, with the possibility to be changed to `:none`, only. If we decide to go this way, which may break some tests, we would need to include few `is_windows` checks to set the defaults. 

What do you think about this?

**Comment by lbenet** [comment]
Fixed by #143; closing.
